ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/ValidatedCommandFactory.java

```diff
@@ -177,6 +187,66 @@ private static Command createForAlterSystemQuery(
     return Command.of(statement);
   }
 
+  private static Command createForPauseQuery(
+      final ConfiguredStatement<? extends Statement> statement,
+      final KsqlExecutionContext context
+  ) {
+    final PauseQuery pauseQuery = (PauseQuery) statement.getStatement();
+    final Optional<QueryId> queryId = pauseQuery.getQueryId();
+
+    if (!queryId.isPresent()) {
+      context.getPersistentQueries().forEach(PersistentQueryMetadata::pause);
```

Comment:
Note for myself (and other readers) since I was confused the first time through: this code here is executed against a sandbox, in order to validate that the query ID exists and is not a source table query. The analogous code in InteractiveStatementExecutor is the one that actually executes this statement from the command topic.

Reply:
+1.  Yeah, some of this is confusing.  Shout if there's an obvious way to improve it for this PR!